src/vllm_tgis_adapter/grpc/grpc_server.py

```diff
@@ -177,12 +177,12 @@ class TextGenerationService(generation_pb2_grpc.GenerationServiceServicer):
 
     def __init__(
         self,
-        engine: AsyncEngineClient | AsyncLLMEngine,
+        engine: EngineClient | AsyncLLMEngine,
```

Comment:
I'm not sure that the union should be needed here.. `EngineClient` is a `Protocol` and `AsyncLLMEngine` implements that protocol.. so only the former should be needed I think?